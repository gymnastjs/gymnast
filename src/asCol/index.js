// @flow
import * as React from 'react'
import asGrid from '../asGrid'
import type { OneResolution } from '../types'

function mapProps(props): OneResolution {
  const defaultProps: OneResolution = {
    marginTop: 0,
    marginRight: 'gutter/2',
    marginBottom: 'verticalGutter',
    marginLeft: 'gutter/2',
  }

  return props.margin !== undefined
    ? props
    : {
        ...defaultProps,
        ...props,
      }
}

export default function asCol(Component: React.ComponentType<*> | string) {
  return asGrid(Component, mapProps)
}
