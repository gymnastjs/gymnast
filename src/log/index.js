// @flow
/* eslint-disable no-console */
import errors from '../errors'

type LogLevels = 'info' | 'warn' | 'error'
const logLevels = ['info', 'warn', 'error']
let logIndex = 0
let logger = console
const log = {}

logLevels.forEach((level, index) => {
  log[level] = (...args: any[]) => {
    if (index >= logIndex) {
      logger[level](...args)
    }
  }
})

log.setLevel = (level: LogLevels) => {
  const index = logLevels.indexOf(level)

  if (index >= 0) {
    logIndex = index
  } else {
    log.error(errors.INVALIDLOGLEVEL, level)
  }
}

log.setLogger = (newLogger: { [LogLevels]: (...args: any[]) => void }) => {
  logger = newLogger
}

export default log
