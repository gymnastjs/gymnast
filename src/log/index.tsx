/* eslint-disable no-console */
import { errors } from '../errors'
import { Logger } from '../types'

const logLevels = ['info', 'warn', 'error']
let logIndex = 0

let logger: Logger = console

const log = {
  info: (...args: any[]) => logIndex <= 0 && logger.info(...args),
  warn: (...args: any[]) => logIndex <= 1 && logger.warn(...args),
  error: (...args: any[]) => logIndex <= 2 && logger.error(...args),
  setLevel(level: 'info' | 'warn' | 'error') {
    const index = logLevels.indexOf(level)

    if (index >= 0) {
      logIndex = index
    } else {
      log.error(errors.INVALIDLOGLEVEL, level)
    }
  },
  setLogger(newLogger: Logger) {
    logger = newLogger
  },
}
export default log
