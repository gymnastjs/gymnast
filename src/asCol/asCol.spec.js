import React from 'react'
import { mount } from 'enzyme'
import log from '../log'
import asCol from './index'

describe('Col', () => {
  let wrapper
  const Col = asCol('div')

  it('should render a Grid with col margins ([0, 1.5, 3])', () => {
    wrapper = mount(<Col />)
    const gridProps = wrapper.find('div').props()
    expect(gridProps).toMatchSnapshot()
  })

  it('should override defaults when passing margin prop', () => {
    wrapper = mount(<Col margin={1} />)
    const gridProps = wrapper.find('Grid').props()
    expect(gridProps).toMatchSnapshot()
  })

  it('should render custom elements', () => {
    const StrongCol = asCol('strong')

    wrapper = mount(<StrongCol />)
    expect(wrapper.find('strong').length).toBe(1)
  })

  it('should not error when passed valid props', () => {
    spyOn(log, 'error')

    wrapper = mount(<Col margin={0} />)

    expect(log.error).not.toHaveBeenCalled()
  })

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount()
    }
  })
})
