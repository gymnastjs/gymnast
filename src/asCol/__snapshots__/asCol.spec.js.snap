// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should override defaults when passing margin prop 1`] = `
Object {
  "context": Object {
    "base": 8,
    "columns": 12,
    "displayAliases": Object {
      "large": Array [
        Object {
          "minWidth": "900px",
        },
        Object {
          "minDeviceWidth": "900px",
        },
      ],
      "medium": Array [
        Object {
          "maxWidth": "899px",
          "minWidth": "600px",
        },
        Object {
          "maxDeviceWidth": "899px",
          "minDeviceWidth": "600px",
        },
      ],
      "small": Array [
        Object {
          "maxWidth": "599px",
        },
        Object {
          "maxDeviceWidth": "599px",
        },
      ],
    },
    "fallbackDisplayKey": "default",
    "gutter": 3,
    "maxPageWidth": 153,
    "minPageWidth": 40,
    "pageMargin": Object {
      "large": 6,
      "medium": 6,
      "small": 1,
    },
    "spacingAliases": Object {
      "2XL": 6,
      "2XL/2": 3,
      "L": 3,
      "L/2": 1.5,
      "M": 2,
      "M/2": 1,
      "S": 1,
      "S/2": 0.5,
      "XL": 4,
      "XL/2": 2,
      "XS": 0.5,
    },
    "verticalGutter": 3,
  },
  "margin": 1,
  "show": undefined,
}
`;

exports[`Col should render a Grid with col margins ([0, 1.5, 3]) 1`] = `
Object {
  "className": "{\\"border\\":\\"0 transparent solid\\",\\"boxSizing\\":\\"border-box\\",\\"display\\":\\"flex\\",\\"flexFlow\\":\\"row wrap\\",\\"width\\":\\"100%\\"} {\\"flexGrow\\":1}",
  "show": undefined,
  "style": Object {
    "borderBottomWidth": 24,
    "borderLeftWidth": 12,
    "borderRightWidth": 12,
    "borderTopWidth": 0,
  },
}
`;
